README.md

```diff
@@ -1,10 +1,114 @@
 # dojo-i18n
 
-TODO: Replace with a description of this package
+An internationalization library that provides locale-specific message loading, and support for locale-specific message, date, and number formatting.
 
 ## Features
 
-TODO: Add sections on features of this package
+The examples below are provided in TypeScript syntax. The package does work under JavaScript, but for clarity, the examples will only include one syntax.
+
+### Message Bundle Loading
+
+`dojo-i18n` provides a means for loading locale-specific messages, and updating those messages when the locale changes. Each bundle has a default module that is `import`ed like any other TypeScript module. Locale-specific messages are then loaded via the `i18n` method. Every default bundle MUST provide a `baseUrl` that will be used to determine locale bundle locations, a `locales` array of supported locales, and a `messages` map of default messages. For example, suppose the module located at `nls/common.ts` contains the following contents:
+
+```typescript
+import { Bundle, Messages } from 'dojo-i18n/i18n';
+
+const baseUrl = 'nls/common';
```

Comment:
is `baseUrl` the right name for this property, as it's not actually the `baseUrl`, for example if I had an nls file at `app/component/nls/widget.ts` I would assume the `baseUrl` would be `app/component/nls/` but it's not. According to the documentation it should be `app/component/nls/widget`... unless I completely misread this.